### PR TITLE
Set writer and date in setItemText method

### PR DESCRIPTION
**Pull request by rookgaard**
### Pull Request Prelude

- [x] I have followed [proper The Forgotten Server code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed

Since you can set the text of the item, it may also be useful for someone to add the author and date, e.g. to a parcel given by the server or to a quest where a letter is given to the player.

[cont]: https://github.com/otland/forgottenserver/wiki/Contributing
[code]: https://github.com/otland/forgottenserver/wiki/TFS-Coding-Style-Guide

### CODE REVIEW

**Review by EPuncker — APPROVED** [review]

File: data/lib/compat/compat.lua [hunk 1]
```diff
@@ -1065,6 +1065,19 @@ function doSetItemText(uid, text)
 	else
 		item:removeAttribute(ITEM_ATTRIBUTE_TEXT)
 	end
+
+	if writer ~= "" then
```
Comment (MillhioreBT) [inline data/lib/compat/compat.lua:1069]: `if writer ~= nil then`

File: data/lib/compat/compat.lua [hunk 2]
```diff
@@ -1065,6 +1065,19 @@ function doSetItemText(uid, text)
...
+		item:setAttribute(ITEM_ATTRIBUTE_WRITER, tostring(writer))
+	else
+		item:removeAttribute(ITEM_ATTRIBUTE_WRITER)
+	end
+
+	if date ~= "" then
```
Comment (MillhioreBT) [inline data/lib/compat/compat.lua:1075]: `if date ~= nil then`

**Review by MillhioreBT — CHANGES_REQUESTED** [review]

File: data/lib/compat/compat.lua (hunk 1 above)
Comment (EPuncker) [reply data/lib/compat/compat.lua:1069]: we removed almost every nil from lua scripts, so I'd suggest:
```suggestion
	if writer then
```

same for date

**Review by EPuncker — CHANGES_REQUESTED** [review]

File: data/lib/compat/compat.lua (hunk 1 above)
Comment (luanluciano93) [reply data/lib/compat/compat.lua:1069]:
```suggestion
	if writer and writer ~= "" then
```

File: data/lib/compat/compat.lua (hunk 2 above)
Comment (luanluciano93) [reply data/lib/compat/compat.lua:1075]:
```suggestion
	if date and date ~= "" then
```

**Review by MillhioreBT — APPROVED** [review]

**Review by EPuncker — APPROVED** [review]

**Pull request merged**